### PR TITLE
Command processing flow

### PR DESCRIPTION
**Pull request by Deseteral**

### CODE REVIEW

File: service/src/main/kotlin/xyz/lebkuchenfm/Application.kt [hunk 1]
```diff
@@ -32,10 +38,22 @@ fun Application.module() {
     val songsRepository = SongsMongoRepository(database)
     val songsService = SongsService(songsRepository)
 
+    val eventStream = DummyEventStream() // TODO: To be replaced with actual WebSocket implementation.
+
+    val commandPrompt = environment.config.property("commandPrompt").getString()
```
Comment (gorzalczany) [inline service/src/main/kotlin/xyz/lebkuchenfm/Application.kt:43]: Maybe for now we should pass it like everywhere else and then make global refactor of config injecting.

File: service/src/main/kotlin/xyz/lebkuchenfm/domain/commands/processors/XCommandProcessor.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        exampleUsages = listOf("airhorn"),
+        parameters = CommandParameters(
+            parameters = listOf(
+                CommandParameters.RequiredCommandParameter("sound-name"),
+            ),
+        ),
+    ) {
+    override fun execute(command: Command): CommandProcessingResult {
+        val soundName = command.rawArgs
+            ?: return error("You have to provide sound name.", logger)
```
Comment (gorzalczany) [inline service/src/main/kotlin/xyz/lebkuchenfm/domain/commands/processors/XCommandProcessor.kt:28]: So we have user facing error in english but help message in polish? 🤔

**Review by gorzalczany — APPROVED** [review]

File: service/src/main/kotlin/xyz/lebkuchenfm/Application.kt [hunk 3]
```diff
@@ -7,13 +7,19 @@ import io.ktor.server.http.content.staticResources
...
 import xyz.lebkuchenfm.api.songs.songsRouting
 import xyz.lebkuchenfm.api.xsounds.xSoundsRouting
+import xyz.lebkuchenfm.domain.commands.CommandExecutorService
+import xyz.lebkuchenfm.domain.commands.CommandProcessorRegistry
+import xyz.lebkuchenfm.domain.commands.TextCommandParser
+import xyz.lebkuchenfm.domain.commands.processors.XCommandProcessor
 import xyz.lebkuchenfm.domain.songs.SongsService
 import xyz.lebkuchenfm.domain.xsounds.XSoundsService
-import xyz.lebkuchenfm.external.storage.mongo.MongoDatabaseClient
+import xyz.lebkuchenfm.external.DummyEventStream
```
[the remark this answers is not in the document]
Comment (Deseteral) [reply service/src/main/kotlin/xyz/lebkuchenfm/Application.kt:19]: This is exactly why it is named that way :D

[the remark this answers is not in the document]
Comment (Deseteral) [reply service/src/main/kotlin/xyz/lebkuchenfm/Application.kt:19]: It's not meant to be permanent, so I wanted to make sure that it isn't.

File: service/src/main/kotlin/xyz/lebkuchenfm/Application.kt (hunk 1 above)
Comment (Deseteral) [reply service/src/main/kotlin/xyz/lebkuchenfm/Application.kt:43]: Configuration is a convention problem that we will have to solve for sure 👍

File: service/src/main/kotlin/xyz/lebkuchenfm/domain/commands/processors/XCommandProcessor.kt (hunk 2 above)
Comment (Deseteral) [reply service/src/main/kotlin/xyz/lebkuchenfm/domain/commands/processors/XCommandProcessor.kt:28]: Oh yes, I wanted to talk to you about it and forgot. First - it should be consistent, but I forgot to change it.

My current thinking is that we will write everything in english (as opposed to polish-english that we currently have on prod) and then tackle i18n after v3 is deployed. How does that sound to you?

File: service/src/main/kotlin/xyz/lebkuchenfm/Application.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (gorzalczany) [reply service/src/main/kotlin/xyz/lebkuchenfm/Application.kt:19]: Ok then
image

**Pull request merged**